run_md.py

```diff
@@ -0,0 +1,879 @@
+    :param clean_previous: boolean. Remove all previous md files
+    :param gromacs_version:
+    :param mdtime_ns: float. Time in ns
+    :param npt_time_ps: int. Time in ps
+    :param nvt_time_ps: int. Time in ps
+    :param topol: None or file
+    :param topol_itp_list: None or list of files
+    :param posre_list_protein: None or list of files
+    :param wdir_to_continue_list: list of paths
+    :param tpr_prev: None or file
+    :param cpt_prev: None or file
+    :param xtc_prev: None or file
+    :param deffnm_prev: md_out
+    :param hostfile: None or file
+    :param ncpu:
+    not_clean_log_files: boolean. Remove backup md files (starts with #)
+    :return:
+    '''
+
+
+    try:
+        subprocess.check_output(f'module load {gromacs_version}', shell=True)
+    except subprocess.CalledProcessError as e:
+        logging.error(e)
+        return None
+
+    project_dir = os.path.dirname(os.path.abspath(__file__))
+    script_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'scripts')
+
+    if wdir_to_continue_list is None:
+        # create dirs
+        wdir_protein = os.path.join(wdir, 'md_preparation', 'protein')
+        wdir_ligand = os.path.join(wdir, 'md_preparation', 'var_lig')
+        wdir_system_ligand = os.path.join(wdir, 'md_preparation', 'system_lig')
+
+        wdir_md = os.path.join(wdir, 'md_preparation', 'md_files')
+
+        os.makedirs(wdir_md, exist_ok=True)
+        os.makedirs(wdir_protein, exist_ok=True)
+        os.makedirs(wdir_ligand, exist_ok=True)
+        os.makedirs(wdir_system_ligand, exist_ok=True)
+
+        if not os.path.isfile(protein):
+            raise FileExistsError(f'{protein} does not exist')
+
+        pname, p_ext = os.path.splitext(os.path.basename(protein))
+        # check if already exist in the working directory
+        if not os.path.isfile(f'{os.path.join(wdir_protein, pname)}.gro') or not os.path.isfile(os.path.join(wdir_protein, "topol.top")):
+            if p_ext != '.gro' or topol is None or posre_list_protein is None:
+                try:
+                    subprocess.check_output(f'gmx pdb2gmx -f {protein} -o {os.path.join(wdir_protein, pname)}.gro -water tip3p -ignh '
+                          f'-i {os.path.join(wdir_protein, "posre.itp")} '
+                          f'-p {os.path.join(wdir_protein, "topol.top")}'
+                          f'<<< {forcefield_num}', shell=True)
+                except subprocess.CalledProcessError as e:
+                    logging.error(e)
+                    return None
+            else:
+                if not os.path.isfile(os.path.join(wdir_protein, os.path.basename(protein))):
+                    shutil.copy(protein, os.path.join(wdir_protein, os.path.basename(protein)))
+                if not os.path.isfile(os.path.join(wdir_protein, 'topol.top')):
+                    shutil.copy(topol, os.path.join(wdir_protein, 'topol.top'))
+                # multiple chains
```

Comment:
it would be reasonable to check the length of both variables (should be identical) if they are not None